src/transform/activation.jl

```diff
@@ -0,0 +1,132 @@
+
+# -------------------------------------------------------------
+# Repurposed from https://github.com/tbreloff/OnlineAI.jl
+# -------------------------------------------------------------
+
+abstract Activation <: Transformation
+
+value(activation::Activation, input::AbstractVector) = Float64[value(activation, Σ) for Σ in input]
+deriv(activation::Activation, input::AbstractVector) = Float64[deriv(activation, Σ) for Σ in input]
+
+function value!(buffer::AbstractVector, activation::Activation, input::AbstractVector)
+    for i in 1:length(buffer)
+        buffer[i] = value(activation, input[i])
+    end
+end
+function deriv!(buffer::AbstractVector, activation::Activation, input::AbstractVector)
+    for i in 1:length(buffer)
+        buffer[i] = deriv(activation, input[i])
+    end
+end
+
+# ----------------------------------------------------------------------------
+
+doc"""
+f(x) = x
+f'(x) = 1
+"""
+immutable IdentityActivation <: Activation end
+value(activation::IdentityActivation, input::Real) = input
```

Comment:
if you change the type of input to the more abstract `Number` then you can make use of dual numbers in the test. Also there is no downside to allow these things as long as we are careful with typestability